Tools/setup/ubuntu.sh

```diff
@@ -194,6 +188,57 @@ if [[ $INSTALL_NUTTX == "true" ]]; then
 	fi
 fi
 
+# Install JAVA for SITL or FastRTPS
+if [[ $INSTALL_RTPS == "true" || $INSTALL_SIM == "true" ]]; then
+  if [[ "${UBUNTU_RELEASE}" == "18.04" ]]; then
+		java_version=11
+	else
+		java_version=14
+	fi
+
+  wget -O /tmp/jdk-public-key https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public \
+    && gpg --no-default-keyring --keyring /tmp/adoptopenjdk-keyring.gpg --import /tmp/jdk-public-key \
+    && gpg --no-default-keyring --keyring /tmp/adoptopenjdk-keyring.gpg --export --output /tmp/adoptopenjdk-archive-keyring.gpg \
+    && rm /tmp/adoptopenjdk-keyring.gpg \
+    && mv /tmp/adoptopenjdk-archive-keyring.gpg /usr/share/keyrings \
+    && echo "deb [signed-by=/usr/share/keyrings/adoptopenjdk-archive-keyring.gpg] https://adoptopenjdk.jfrog.io/adoptopenjdk/deb focal main" | sudo tee /etc/apt/sources.list.d/adoptopenjdk.list
+
+  sudo DEBIAN_FRONTEND=noninteractive apt-get update && apt-get -y --quiet --no-install-recommends install \
+    adoptopenjdk-$java_version-hotspot
+fi
+
+# Fast-RTPS
+if [[ $INSTALL_RTPS == "true" ]]; then
+
+  wget -O /tmp/gradle-6.4.1-bin.zip https://services.gradle.org/distributions/gradle-6.4.1-bin.zip \
+    && unzip -d /opt/gradle /tmp/gradle-6.4.1-bin.zip
+  export PATH="$PATH:/opt/gradle/gradle-6.4.1/bin"
+
+  # Intall foonathan_memory from source as it is required to Fast-RTPS >= 1.9
+  git clone https://github.com/eProsima/foonathan_memory_vendor.git /tmp/foonathan_memory \
+    && cd /tmp/foonathan_memory \
+    && mkdir build && cd build \
+    && cmake .. \
+    && cmake --build . --target install -- -j $(nproc)
+
+  # Fast-DDS (Fast-RTPS 2.0.0)
+  git clone --recursive https://github.com/eProsima/Fast-DDS.git -b v2.0.2 /tmp/FastRTPS-2.0.2 \
+    && cd /tmp/FastRTPS-2.0.2 \
+    && mkdir build && cd build \
+    && cmake -DTHIRDPARTY=ON -DSECURITY=ON .. \
+    && cmake --build . --target install -- -j $(nproc)
+
+  # Fast-RTPS-Gen 1.0.4
+  git clone --recursive https://github.com/eProsima/Fast-DDS-Gen.git -b v1.0.4 /tmp/Fast-RTPS-Gen-1.0.4 \
+    && cd /tmp/Fast-RTPS-Gen-1.0.4 \
+    && gradle assemble \
+    && gradle install
+
+  # cleanup installation
+  rm -rf /tmp/*
```

Comment:
```suggestion
```

This is more of a Dockerfile thing we shouldn't be doing directly on regular systems.

Reply:
I was actually thinking we could just remove the saved files we placed in there instead of removing everything, but if we are not concerned with that we can just leave them